bundles/org.openhab.binding.enocean/src/main/java/org/openhab/binding/enocean/internal/eep/EEPFactory.java

```diff
@@ -51,7 +52,7 @@ public class EEPFactory {
     public static EEP createEEP(EEPType eepType) {
         try {
             Class<? extends EEP> cl = eepType.getEEPClass();
-            if (cl == null) {
+            if (Objects.isNull(cl)) {
```

Comment:
I wanted to lower the amount of warnings on building and with the code before I was getting
```
[WARNING] /home/zhivka/workspace/openhab/openhab-addons/bundles/org.openhab.binding.enocean/src/main/java/org/openhab/binding/enocean/internal/eep/EEPFactory.java:[54,17] Null comparison always yields false: The variable cl cannot be null at this location
[WARNING] /home/zhivka/workspace/openhab/openhab-addons/bundles/org.openhab.binding.enocean/src/main/java/org/openhab/binding/enocean/internal/eep/EEPFactory.java:[54,29] Dead code
```
After my change those warnings didn't appear. Do you know better way to fix those warnings?

Reply:
According to the `getEEPClass()` annotation it will never return null, so the compiler knows this is dead code.
So either these checks can be removed OR the annotation should be changed. I'm 99% sure its the first.